tcp_endpoint/src/tcp_endpoint/RosTCPClientThread.py

```diff
@@ -130,14 +133,27 @@ def run(self):
             print("No data for a message size of {}, breaking!".format(full_message_size))
             return
 
-        if destination not in self.source_destination_dict.keys():
-            error_msg = "Topic/Service destination '{}' does not exist in source destination dictionary {} "\
-                .format(destination, self.source_destination_dict.keys())
+        if destination.startswith('__'):
+            if destination not in self.tcp_server.special_destination_dict.keys():
+                error_msg = "System message '{}' is undefined! Known system calls are: {}"\
+                    .format(destination, self.tcp_server.special_destination_dict.keys())
+                self.conn.close()
+                self.tcp_server.send_unity_error(error_msg)
+                raise TopicOrServiceNameDoesNotExistError(error_msg)
```

Comment:
What problem is the special destination dict trying to solve? Is there a reason the `RosUnityHandshakeService` needs to be abstracted like this?

Reply:
My intent is that this is our channel for sending commands directly to the tcp_endpoint, rather than sending general messages to ROS. For example this task https://jira.unity3d.com/browse/AIRO-149 might add a List All Topics Service to this dictionary.

Reply:
I see... so the double underscore indicates a sort of "Do Not Forward" message that is meant to be interpreted by the Endpoint rather than sent over ROS? Does it need to follow the same code path as messages that get transparently passed?

Reply:
Not necessarily? I'm trying to figure out the cleanest solution.